docs/proposals/policy.md

```diff
@@ -0,0 +1,146 @@
+// That means the full set of PolicyRuleBindings for a namespaced request are contained inside of the requested 
+// namespace or in the global namespace
+type PolicyRuleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	Rule PolicyRule
+}
+
+// Role is a logical grouping of PolicyRules that can be referenced as a unit by RoleBindings.
+type Role struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	Rules []PolicyRule
+}
+
+// RoleBinding references a Role, but not contain it.  It adds who and namespace information like a PolicyRuleBinding.  
+// It can reference any Role in the same namespace or in the global namespace.
+type RoleBinding struct {
+	kapi.TypeMeta
+	kapi.ObjectMeta
+
+	UserNames    []string
+	TeamNames    []string
+	AccountNames []string
+
+	// this is a ref to a Role
+	ReferencesGlobalRole bool
+	RoleName             string
+}
+```
+
+### What is a global namespace?
+The types used for policy configuration need to be namespaced types to enable efficient retrieval during policy evaluation, but some Policies, Roles, and their associated bindings need to be globally accessable as well.  One way to support this is to create a special global namespace called "MotherBrain".  It would be a reserved, well-known name that would allow policy to be written against it.
+
+### Common Usage
+OpenShift will ship with some default Roles specified in the global namespace.  These will include roles like: admin, manager, developer, and operator.  Nothing except policy precludes a user from creating additional roles or modifying the roles we provide.  An admin can then create a project and start assigning roles to users by creating a RoleBinding.  The PolicyRuleBindings can then be used to fine-tune permissions on a user-by-user basis.  This makes it easy to express ideas like:  I want David and Jordan to have developer access, but I also want Jordan to be able to edit DeploymentConfig objects.  The policy config types will also be protected by the same policy engine.
+
+### How are Roles different than Teams?
+At first glance, having both roles and teams appears superfluous.  What is the difference between binding PolicyRules to a Team and assigning a user to that team and adding PolicyRules to a RoleTemplate and then binding the RoleTemplate to a Team?  The differences lie in the mutability of the rules, traceability in the UI, and efficiency in the storage/retrieval.
+
+Let's use the example of "reader" role.  You have three users: "herriot", "bubba", and "macdonald", two teams: "vets" and "mechanics", and you're trying to protect three different namespaces: "cows", "sheep", and "tractors".  You have a PolicyRule "readingRule" that looks like this: PolicyRule{Verbs: {"watch", "list", "get"}, ResourceKind: "*"}.  Vets" should be able to read inside of "cows" and "sheep", but not "tractors".  "Mechanics" should be able to read "tractors", but not "cows" or "sheep".
```

Comment:
When are admins found to define policies across multiple namespaces?  I don't see that as very common - it's more likely an admin wants to assign the same policy to multiple namespaces, and either modifies the shared policy, or clones it and mutates their new copy.  The objective of flexible policy is to give more control over permissions - not to mutate policy in bulk.  We do not need to have a generic policy engine - instead we need to design the policy options we need to support.  I don't think its a use case to modify a single policy and add or remove namespaces it controls, because the project is ultimately the resource that policy derives from.  The policy rules need to be set up so that admins can control policy mutation within a project, and control creation of a new project and the modification of what policies are applied to the project.